lld/COFF/InputFiles.cpp

```diff
@@ -660,10 +660,15 @@ std::optional<Symbol *> ObjFile::createDefined(
 
     if (prevailing) {
       SectionChunk *c = readSection(sectionNumber, def, getName());
-      sparseChunks[sectionNumber] = c;
-      c->sym = cast<DefinedRegular>(leader);
```

Comment:
nit: you could also insert at L664:
```
      if (!c)
        return nullptr;
```

Reply:
Updated it that way; thanks, that looks even neater!